### DISCUSSION TITLE
Recommendation for docTR on CPU for Multilingual OCR

### DISCUSSION THREAD

**Discussion by mkrzywda** [mindee/doctr#1931] [Q&A]
Hi everyone,  

I’m currently using docTR for OCR on GPU with the following configuration:  

```python
DET_ARCH = "fast_tiny"
REC_ARCH = "crnn_mobilenet_v3_small"
BIN_THRESH = 0.1
BOX_THRESH = 0.1    

model = ocr_predictor(det_arch=DET_ARCH, reco_arch=REC_ARCH, pretrained=True).to(device)                                           
doc = DocumentFile.from_images([tmp_path])
result = model(doc)

```

It works well, but I’m looking to switch to CPU and need recommendations for the best setup. My requirements are:

- Multilingual support: Mostly English, occasionally Polish.
- Low processing time: For images up to 4K resolution.
- Bounding box per word: High precision is crucial.

Any suggestions for optimal `DET_ARCH`, `REC_ARCH`, or other parameters to balance speed and accuracy on CPU?

Thanks!

**Comment by felixdittrich92** [comment]
Hi @mkrzywda :wave:,

with all the requirements you should check out the following:

- [OnnxTR](https://github.com/felixdittrich92/OnnxTR)  latency benchmarks: https://github.com/felixdittrich92/OnnxTR#benchmarks
- multilingual (latin extended):
   - docTR: https://huggingface.co/Felix92/doctr-torch-parseq-multilingual-v1
   - OnnxTR: https://huggingface.co/Felix92/onnxtr-parseq-multilingual-v1
   
   
Hope this helps

Best,
Felix

**Reply by felixdittrich92** [reply]
Dataset I used to train this model can be downloaded here: 
https://drive.google.com/file/d/1TNQN8uBMiGjzf2GM41BWDICefLubau5q/view?usp=sharing

Currently working on a better and more extended dataset

**Reply by mkrzywda** [reply]
What is a key-diff beetwen `docTR` and `OnnxTR` ? Because in `docTR` I can use more models than in `OnnxTR` ?

**Reply by felixdittrich92** [reply]
OnnxTR and docTR provides the same models(https://github.com/felixdittrich92/OnnxTR#text-detection), the key difference is that OnnxTR is a core refactoring of docTR which works with `onnxruntime` as backend instead of `PyTorch` or `TensorFlow`. 
`onnxruntime` is hardware optimized, faster and more lightweight in general, additional it provides easy support for different accelerators - besides CPU / GPU (CUDA) also OpenVNO, DirectML, etc.

So the user interface between OnnxTR & docTR is mostly the same but the core differs - one difference: OnnxTR is inference only but optimized especially for prod systems  and in docTR you have both inference & training logic

**Reply by mkrzywda** [reply]
Wonderful! I'm excited to try `OnnxTR` on CPU for processing images in this scenario. Is it possible to fine-tune models for my specific type of images?

**Reply by felixdittrich92** [reply]
Yes in this case use docTR to finetune the model/models to your own needs
1. https://mindee.github.io/doctr/using_doctr/custom_models_training.html
then you can export the model to onnx
2. https://mindee.github.io/doctr/using_doctr/using_model_export.html#export-to-onnx
and afterwards load in OnnxTR
3. https://github.com/felixdittrich92/OnnxTR#loading-custom-exported-models

:)
